client/src/pages/HomePage/HomePage.js

```diff
@@ -1,10 +1,64 @@
-import React from 'react';
+import React, { useContext, useState, useEffect } from 'react';
+import axios from 'axios';
+import { StateContext } from '../../App';
+import { Box, Typography } from '@mui/material';
+import AnimalTypeBox from './components/AnimalTypeBox';
+import AddAnimal from './components/AddAnimal';
+import {
+  homePageContainer,
+  animalTypeContainer,
+  titleText,
+} from '../../styles/home/animalTypeBoxStyles';
 
 const HomePage = () => {
+  const { state } = useContext(StateContext);
+  const [animalTypes, setAnimalTypes] = useState([]);
+
+  const axiosConfig = {
+    headers: {
+      Authorization: `Bearer ${state.token}`,
+    },
+  };
+
+  const fetchData = async () => {
+    try {
+      const response = await axios.get(
+        'http://localhost:5000/api/v1/animals',
+        axiosConfig
+      );
+      const data = response.data.animalList;
+      setAnimalTypes(data);
+    } catch (error) {
+      console.error(error.response);
+    }
+  };
+
+  const addAnimal = async (newAnimal) => {
+    const { name, type } = newAnimal;
```

Comment:
put a checker here if `null` or `undefined` ni nga mga values kay required mani sila nga mga params, because even tho ga check nana sa backend, mas better if ang both source kay ga check sa mga inputs

Reply:
resolved, but instead of `null` or `undefined` akong gi check, akoang gi `if(name && type)`

Reply:
resolved